src/remote.c

```diff
@@ -670,18 +670,26 @@ int git_remote_set_pushurl(git_repository *repo, const char *remote, const char*
 	return set_url(repo, remote, CONFIG_PUSHURL_FMT, url);
 }
 
-const char* git_remote__urlfordirection(git_remote *remote, int direction)
+static const char *git_remote__urlresolve(const char *url, int direction, const git_remote_callbacks *callbacks)
+{
+    if (callbacks != NULL && callbacks->resolve_url != NULL) {
+        return callbacks->resolve_url(url, direction, callbacks->payload);
+    }
+    return url;
+}
+
+const char* git_remote__urlfordirection(git_remote *remote, int direction, const git_remote_callbacks *callbacks)
```

Comment:
minor nit: we put the `*` on the name not on the type.  eg `const char *git_remote__`...

Reply:
yes, i noticed, but it already was that way (not my code) so i left it on the type.